packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -93,51 +132,32 @@ function NavigationMenuItemEdit( {
 						name="link"
 						icon="admin-links"
 						title={ __( 'Link' ) }
-						onClick={ () => setIsLinkOpen( ! isLinkOpen ) }
+						onClick={ () => {
+							// If the popover was closed by click outside,
+							// then there is not nothing to do here.
+							if ( wasClosedByLinkControl ) {
```

Comment:
Would it be possible to simplify this logic a bit and remove the wasClosedByLinkControl.
What I had in mind is:
- Unselecting the block makes isLinkOpen false (an effect like we have now).
- onClose also sets isLinkOpen to false.
- The onClick of the ToolbarButton would only be passed if the toolbar was closed and would set  isLinkOpen to true, if the link was open we would pass noop for the onClick as it does not needs to do anything, the onClose of the popover will set isLinkOpen to false to close the toolbar.

Do you think this would work?

In another point currently we have a small bug:
Focus the text in a menu item.
Press the link button to open the popover.
Focus the text in the same menu item again. The popover closes.
Press the link burron to open the popover. The popover does not opens when we click. If we click again it opens.

Reply:
yes, there is a gif above with this bug visually shown

Reply:
> Do you think this would work?

I don't think so :-(

It's a matter of which event triggers first. In this case, the `Popover.onFocusOutside` wins to the `ToolbarButton.onClick`, so the popover always is closed when the onClick happens/
Going to implement a hacky solution.

Reply:
this fec622aa1c8ce8ca0c163e055b7bf049f458ff0a

Reply:
I've created a [different approach](bd8bbf5fe122ed34f09237bfd32bc8932aa8d9c9). it's less hacky but [requires to touch the toolbar button](e01f4bac7aef1842bcb780dea6dc94540b31b2d9), is not a big deal, though.